telemetry/telemetry.go

```diff
@@ -0,0 +1,111 @@
+// As part of the White Glove White Glove project,
+// this is a telemetry package to support posting summary and cancellation events
+// to a given webhook endpoint with a reference project id in the shape of an event grid
+// custom event.
+
+package telemetry
+
```

Comment:
Not sure where to put this, but we have a PR in the works regarding logging to Azure Monitor.

Would this serve the intended purpose here, or is there a special need to log to your service?

Reply:
Furthermore, this PR in general could use a pretty strong refactor in general.

First of all, consider whether the aforementioned PR may better fit your use case.
Considering we already have a logger in place, consider moving the entirety of this into that architecture.
Furthermore, Consider relegating the aforementioned parameters to environment variables.
If you need logging in the frontend regarding certain events, we do have a workaround to do that. Check out JobsAdmin.LogToJobLog(string). Note that this can overflow at 1000 items, and when it does, further logs will be dropped.

---
In reply to: [345364952](https://github.com/Azure/azure-storage-azcopy/pull/700#discussion_r345364952) [](ancestors = 345364952)

Reply:
@adreed-msft  comments about xyz "in general" are not very actionable. Was there anything you were thinking of other than specific points you mentioned?

Reply:
@msalemor If I recall correctly from what you said, the planned logged to Azure Monitor would not help in your case, because in your case the team receiving the logs (e.g. your team) may be monitoring a job on behalf of a customer.  So if the log goes to Azure Monitor, _in the customer's account_, that doesn't help you to monitor its progress for them, because you can't see Azure Monitor logs in their account. Is that correct?  I vaguely recall some conversation about this, but we need to check/confirm the details to decide whether this feature is indeed logically separate from the logs-to-Azure-Monitor feature.